tmo-monitor.py

```diff
@@ -134,79 +150,217 @@ def sha256url(self, val1, val2):
+    args = self.parse_commandline()
+    self.parse_arguments(args)
+
+    if self.skip_reboot and self.reboot_now:
+      print('Incompatible options: --reboot and --skip-reboot\n', file=sys.stderr)
+      self.parser.print_help(sys.stderr)
+      sys.exit(2)
+    if self.skip_reboot:
+      for var in {'ping', '4G_band', '5G_band', 'enbid'}:
+        self.reboot[var] = False
+    # if not self.login['username']:
+    #   print('The following argument is required: username')
+    #   sys.exit(2)
+    if not self.login['password']:    
+      self.password = getpass.getpass('Password: ')
+
+
+  def read_environment(self):
+    try:
+      envfile=find_dotenv()
+      load_dotenv(envfile)
+    except:
+      logging.debug("No .env file found")
+      return
+    for var in {'username', 'password'}:
+      tmp = os.environ.get('tmo_' + var)
+      if tmp != None:
+        self.login[var] = tmp
+    for var in {'interface', 'ping_host', 'ping_count', 'ping_interval'}:
+      tmp = os.environ.get('tmo_' + var)
+      if tmp != None:
+        self.ping[var] = tmp  
+    for var in {'primary_band', 'secondary_band'}:
+      tmp = os.environ.get('tmo_' + var)
+      if tmp != None:
+        splits = tmp.split(',')
+        self.connection[var] = splits
+    tmp = os.environ.get('tmo_enbid')
+    if tmp != None:
+      self.connection['enbid'] = tmp
+    for var in {'uptime', 'ping', '4G_band', '5G_band', 'enbid'}:
+      tmp = os.environ.get('tmo_' + var + '_reboot')
+      if tmp != None:
+        if tmp.lower() == 'true':
+          self.reboot[var] = True
+        else:
+          self.reboot[var] = False
+    tmp = os.environ.get('tmo_skip_reboot')
+    if tmp != None:
+      self.skip_reboot = tmp
+    tmp = os.environ.get('tmo_print_config')
+    if tmp != None:
+        if tmp.lower() == 'true':
+          self.general['print_config'] = True
+        else:
+          self.general['print_config'] = False
+     
+    
+  def parse_commandline(self):
+    self.parser = argparse.ArgumentParser(description='Check T-Mobile Home Internet cellular band(s) and connectivity and reboot if necessary')
+    # login settings
+    self.parser.add_argument('username', type=str, help='the username (most likely "admin")', nargs='?')
+    self.parser.add_argument('password', type=str, help='the administrative password (will be requested at runtime if not passed as argument)', nargs='?')
+    # ping configuration 
+    self.parser.add_argument('-I', '--interface', type=str, help='the network interface to use for ping. pass the source IP on Windows')
+    self.parser.add_argument('-H', '--ping-host', type=str, default='google.com', help='the host to ping (defaults to google.com)')
+    self.parser.add_argument('--ping-count', type=int, default='1', help='how many ping health checks to perform')
+    self.parser.add_argument('--ping-interval', type=int, default='10', help='how long in seconds to wait between ping health checks')
+    #reboot settings
```

Comment:
inconsistent comment spacing
```suggestion
    # reboot settings
```

Reply:
fixed, thanks